test/e2e/petset.go

```diff
@@ -243,6 +243,120 @@ var _ = framework.KubeDescribe("PetSet [Slow] [Feature:PetSet]", func() {
+		"foo": "bar",
+		"baz": "blah",
+	}
+	headlessSvcName := "test"
+	podName := "test-pod"
+	petSetName := "web"
+	petPodName := "web-0"
+
+	BeforeEach(func() {
+		framework.SkipUnlessProviderIs("gce", "vagrant")
+		By("creating service " + headlessSvcName + " in namespace " + f.Namespace.Name)
+		headlessService := createServiceSpec(headlessSvcName, "", true, labels)
+		_, err := f.Client.Services(f.Namespace.Name).Create(headlessService)
+		framework.ExpectNoError(err)
+		c = f.Client
+		ns = f.Namespace.Name
+	})
+
+	AfterEach(func() {
+		if CurrentGinkgoTestDescription().Failed {
+			dumpDebugInfo(c, ns)
+		}
+		By("Deleting all petset in ns " + ns)
+		deleteAllPetSets(c, ns)
+	})
+
+	It("should recreate evicted petset", func() {
+		By("looking for a node to schedule pet set and pod")
+		nodes := framework.GetReadySchedulableNodesOrDie(f.Client)
+		node := nodes.Items[0]
+
+		By("creating pod with conflicting port in namespace " + f.Namespace.Name)
+		conflictingPort := api.ContainerPort{HostPort: 21017, ContainerPort: 21017, Name: "conflict"}
+		pod := &api.Pod{
+			ObjectMeta: api.ObjectMeta{
+				Name: podName,
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:  "nginx",
+						Image: "gcr.io/google_containers/nginx-slim:0.7",
+						Ports: []api.ContainerPort{conflictingPort},
+					},
+				},
+				NodeName: node.Name,
+			},
+		}
+		pod, err := f.Client.Pods(f.Namespace.Name).Create(pod)
+		framework.ExpectNoError(err)
+
+		By("creating petset with conflicting port in namespace " + f.Namespace.Name)
+		petMounts := []api.VolumeMount{{Name: "datadir", MountPath: "/data/"}}
+		podMounts := []api.VolumeMount{{Name: "home", MountPath: "/home"}}
+		ps := newPetSet(petSetName, f.Namespace.Name, headlessSvcName, 1, petMounts, podMounts, labels)
+		petContainer := &ps.Spec.Template.Spec.Containers[0]
+		petContainer.Ports = append(petContainer.Ports, conflictingPort)
+		ps.Spec.Template.Spec.NodeName = node.Name
+		_, err = f.Client.Apps().PetSets(f.Namespace.Name).Create(ps)
+		framework.ExpectNoError(err)
+
+		By("waiting until pod " + podName + " pet pod " + petPodName + " will be created in namespace " + f.Namespace.Name)
+		Eventually(func() error {
+			_, err := f.Client.Pods(f.Namespace.Name).Get(petPodName)
+			if err != nil {
+				return err
+			}
+			_, err = f.Client.Pods(f.Namespace.Name).Get(podName)
+			if err != nil {
+				return err
+			}
+			return nil
+		}, 2*time.Minute, 2*time.Second).Should(BeNil())
+
+		petPod, err := f.Client.Pods(f.Namespace.Name).Get(petPodName)
+		framework.ExpectNoError(err)
+		initialPetPodUID := petPod.UID
+
+		By("waiting until pet pod will enter failed state in namespace " + f.Namespace.Name)
+		Eventually(func() error {
+			petPod, err := f.Client.Pods(f.Namespace.Name).Get(petPod.Name)
+			// if err not nil - pod was already removed and we can move on to the next step
+			if petPod.Status.Phase == api.PodFailed || err != nil {
+				return nil
+			}
```

Comment:
please describe what we expect to happen in a comment. This is going to race with the petset controller which deletes/recreates pods in parallel right? overall it might work out because of the polling, but feels sketchy. 

We want to prove that in 2minutes the pod enteres failed, and gets recreatd by the petset controller. A more exact way to do this would be via watch (either use `watch.until` or setup your own watch like eg test/e2e/daemon_restart.go), you will get both a delete event, and a recreate for the new pet. 

maybe its sufficient to just poll till the uid changes at least once?